test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -143,6 +142,125 @@ function GetTestData
     return $body
 }
 
+<#
+    Extract the raw content from a response and
+    convert the embedded json to PSObjects.
+    Assumes the raw content is from the  HTTPListener.
+#>
+function Convert-ResponseRawContent
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNull()]
+        [Microsoft.PowerShell.Commands.WebResponseObject] $Response
+    )
+
+    $index = $Response.RawContent.IndexOf("{")
+    [string] $rawContent = $Response.RawContent.SubString($index)
+    return ($rawContent | ConvertFrom-Json)
+}
+
+function ExecuteRedirectRequest
+{
+    param (
+        [Parameter(Mandatory)]
+        [string]
+        $uri,
+
+        [switch] $UsePost,
+
+        [switch] $PreserveAuthorizationOnRedirect
+    )
+    $result = [PSObject]@{Output = $null; Error = $null; Content = $null}
+
+    try
+    {
+        $headers = @{"Authorization" = "test"}
+        if ($UsePost)
+        {
+            $result.Output = Invoke-WebRequest -Uri $uri -TimeoutSec 5 -Headers $headers -PreserveAuthorizationOnRedirect:$PreserveAuthorizationOnRedirect.IsPresent -Method POST
+        }
+        else
+        {
+            $result.Output = Invoke-WebRequest -Uri $uri -TimeoutSec 5 -Headers $headers -PreserveAuthorizationOnRedirect:$PreserveAuthorizationOnRedirect.IsPresent
+        }
+
+        $result.Content = Convert-ResponseRawContent -Response $result.Output
+    }
+    catch
+    {
+        $result.Error = $_
+    }
+
+    return $result
+}
+
+Describe "Invoke-WebRequest redirect tests" -Tags ("Feature", "Redirect") {
```

Comment:
Is there a specific reason you want to add a new Describe rather than having your tests part of the existing Invoke-WebRequest tests?